File: coding-exercises/Leet-Code/1.Two-Sum/Python/1.two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
                      
        mapping = {n: i for i, n in enumerate(nums)}
        
        for i, num in enumerate(nums):
            diff = mapping.get(target - num, False)
            
            if diff and diff != i:
                return [i, diff]

File: coding-exercises/Leet-Code/1.Two-Sum/Javascript/1.two_sum.js
var twoSum = function(nums, target) {
    let obj = {}
    let len = nums.length;
    for(let i = 0;i < len; i++) {
        let containKey = target - nums[i];
        if(obj[containKey]!=undefined) {
            return [obj[containKey],i]
        }
        obj[nums[i]] = i;
    }
};